app/com/linkedin/drelephant/tuning/engine/MRExecutionEngine.java

```diff
@@ -0,0 +1,458 @@
+        String parentParamName = "pig.maxCombinedSplitSize";
+        if (jobSuggestedParamValueMap.containsKey(parentParamName)) {
+          paramValue = jobSuggestedParamValueMap.get(parentParamName);
+        }
+      }
+
+      if (paramValue != null) {
+        JobSuggestedParamValue jobSuggestedParamValue = new JobSuggestedParamValue();
+        jobSuggestedParamValue.paramValue = paramValue;
+        jobSuggestedParamValue.tuningParameter = derivedParameter;
+        jobSuggestedParamValues.add(jobSuggestedParamValue);
+      }
+    }
+  }
+
+  /**
+   * Check if the parameters violated constraints
+   * Constraint 1: sort.mb > 60% of map.memory: To avoid heap memory failure
+   * Constraint 2: map.memory - sort.mb < 768: To avoid heap memory failure
+   * Constraint 3: pig.maxCombinedSplitSize > 1.8*mapreduce.map.memory.mb
+   * @param jobSuggestedParamValueList List of suggested param values
+   * @return true if the constraint is violated, false otherwise
+   */
+
+  @Override
+  public Boolean isParamConstraintViolatedIPSO(List<JobSuggestedParamValue> jobSuggestedParamValueList) {
+    logger.info(" Constraint Violeted ");
+    Double mrSortMemory = null;
+    Double mrMapMemory = null;
+    Double mrReduceMemory = null;
+    Double mrMapXMX = null;
+    Double mrReduceXMX = null;
+    Double pigMaxCombinedSplitSize = null;
+    Integer violations = 0;
+
+    for (JobSuggestedParamValue jobSuggestedParamValue : jobSuggestedParamValueList) {
+      if (jobSuggestedParamValue.tuningParameter.paramName.equals(
+          CommonConstantsHeuristic.ParameterKeys.SORT_BUFFER_HADOOP_CONF.getValue())) {
+        mrSortMemory = jobSuggestedParamValue.paramValue;
+      } else if (jobSuggestedParamValue.tuningParameter.paramName.equals(
+          CommonConstantsHeuristic.ParameterKeys.MAPPER_MEMORY_HADOOP_CONF.getValue())) {
+        mrMapMemory = jobSuggestedParamValue.paramValue;
+      } else if (jobSuggestedParamValue.tuningParameter.paramName.equals(
+          CommonConstantsHeuristic.ParameterKeys.REDUCER_MEMORY_HADOOP_CONF.getValue())) {
+        mrReduceMemory = jobSuggestedParamValue.paramValue;
+      } else if (jobSuggestedParamValue.tuningParameter.paramName.equals(
+          CommonConstantsHeuristic.ParameterKeys.MAPPER_HEAP_HADOOP_CONF.getValue())) {
+        mrMapXMX = jobSuggestedParamValue.paramValue;
+      } else if (jobSuggestedParamValue.tuningParameter.paramName.equals(
+          CommonConstantsHeuristic.ParameterKeys.REDUCER_HEAP_HADOOP_CONF.getValue())) {
+        mrReduceXMX = jobSuggestedParamValue.paramValue;
+      } else if (jobSuggestedParamValue.tuningParameter.paramName.equals(
+          CommonConstantsHeuristic.ParameterKeys.PIG_SPLIT_SIZE_HADOOP_CONF.getValue())) {
+        pigMaxCombinedSplitSize = jobSuggestedParamValue.paramValue / FileUtils.ONE_MB;
+      }
+    }
+
+    if (mrSortMemory != null && mrMapMemory != null) {
```

Comment:
Can we use constants instead of numerical values? And a comment detailing why the following constraints were chosen